bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -322,6 +323,42 @@ public void resumeMinorGC() {
         }
     }
 
+    /**
+     * gc ledger storage.
+     */
+    void gc() {
+        LOG.debug("Starting garbage collecting ledger storage.");
+
+        LOG.debug("Garbage collecting deleted ledgers' index files.");
+        // gc inactive/deleted ledgers
+        doGcLedgers();
+
+        Stopwatch scanStopwatch = new Stopwatch();
+        scanStopwatch.start();
+
+        LOG.debug("Scanning entry log files to extract metadata and delete empty entry logs if possible.");
+        // Extract all of the ledger ID's that comprise all of the entry logs
+        // (except for the current new one which is still being written to).
+        entryLogMetaMap = extractMetaFromEntryLogs(entryLogMetaMap);
+
+        // if it isn't running, break to not access zookeeper
+        if (!running) {
+            return;
+        }
+
+        long elapsedTimeOnScan = scanStopwatch.elapsedMillis();
+
+        if (elapsedTimeOnScan >= gcWaitTime) {
```

Comment:
yes. if the extract/scanning time is longer than gcWaitTime, we gc ledgers again. because during this period, ledgers might be already deleted.